compiler/src/dotty/tools/dotc/transform/TreeGen.scala

```diff
@@ -23,4 +23,15 @@ object TreeGen {
       .select(wrapArrayMethodName(elemtp))
       .appliedToTypes(if (elemtp.isPrimitiveValueType) Nil else elemtp :: Nil)
       .appliedTo(tree)
+
+  /** Determines whether the given `tree` an application of `wrapRefArray` or `wrapXArray` */
+  def isRefOrPrimWrappedArray(tree: Tree, elemTpe: Type)(implicit ctx: Context): Boolean = {
+    tree match {
+      case Apply(fun, _) =>
+        val sname = fun.symbol.name
+        sname == nme.wrapRefArray || sname == nme.wrapXArray(elemTpe.classSymbol.name)
```

Comment:
You can try something like:
```scala
private def isWrapArrayMethodCall(tree: Apply)(implicit ctx: Context): Boolean = {
  val elemType = tree.tpe.elemType
  val wrapArrayMethodSym = defn.ScalaPredefModuleRef.requiredMethodRef(wrapArrayMethodName(elemTpe))
  tree.fun.symbol == wrapArrayMethodSym
}
```
Also I would move this method to `ElimRepeated`

Reply:
I would store the result of `requiredMethodRef` in a val in https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/Definitions.scala (see existing examples like https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/Definitions.scala#L339-L340)

Reply:
Though here it's more complicated since there's one method per primitive type + Ref. You could store all of them in a Set in Definitions and then just check if the current symbol is contained in the Set.